archetype/mustache-template-engine/src/main/java/module-info.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.helidon.build.archetype.mustache.template.engine.MustacheTemplateEngine;
+
+/**
+ * Mustache template engine provider.
+ */
+module mustache.template.engine {
+    requires io.helidon.build.archetype.engine.spi;
+    requires com.github.mustachejava;
+    provides io.helidon.build.archetype.engine.spi.TemplateEngine with MustacheTemplateEngine;
```

Comment:
Can you also add the META-INF/services file. The module-info is only used when module-path is in play.